sdks/python/apache_beam/transforms/trigger.py

```diff
@@ -260,6 +262,55 @@ def to_runner_api(self, unused_context):
         default=beam_runner_api_pb2.Trigger.Default())
 
 
+class AfterProcessingTime(TriggerFn):
+  """Fire exactly once after a specified delay from processing time.
+
+  AfterProcessingTime is experimental. No backwards compatibility guarantees.
+  """
+
+  def __init__(self, delay=0):
+    self.delay = delay
+
+  def __repr__(self):
+    return 'AfterProcessingTime(delay=%d)' % self.delay
+
+  def on_element(self, element, window, context):
+    context.set_timer(
+        '', TimeDomain.REAL_TIME, context.get_current_time() + self.delay)
+
+  def on_merge(self, to_be_merged, merge_result, context):
+    # timers will be kept through merging
+    pass
+
+  def should_fire(self, timestamp, window, context, time_domain):
+    if (time_domain == TimeDomain.REAL_TIME
+        and context.get_current_time() > timestamp):
```

Comment:
<!--new_thread; commit:a68ceca8046b6d42f941aa4016a528ec2ee3808a; resolved:0-->
This condition is not correct--we will always have that for real time, the timestamp will be equal to the current time (or approximately so).  Can you just make this condition `time_domain == TimeDomain.REAL_TIME`?  Add a comment that the real time notification comes from the real time timer set above.

Reply:
Done